integration-tests/tests/tests/modifiers.rs

```diff
@@ -0,0 +1,134 @@
+use cargo_near_integration_tests::generate_abi_fn;
+use function_name::named;
+use std::fs;
+
+#[test]
+#[named]
+fn test_view_function() -> anyhow::Result<()> {
+    let abi_root = generate_abi_fn! {
+        pub fn add(&self, a: u32, b: u32) -> u32 {
+            a + b
+        }
+    };
+
+    assert_eq!(abi_root.abi.functions.len(), 1);
+    let function = &abi_root.abi.functions[0];
+    assert!(function.is_view);
+
+    Ok(())
+}
+
+#[test]
+#[named]
+fn test_call_function() -> anyhow::Result<()> {
+    let abi_root = generate_abi_fn! {
+        pub fn add(&mut self, a: u32, b: u32) -> u32 {
+            a + b
+        }
+    };
+
+    assert_eq!(abi_root.abi.functions.len(), 1);
+    let function = &abi_root.abi.functions[0];
+    assert!(!function.is_view);
+
+    Ok(())
+}
+
+#[test]
+#[named]
+fn test_non_init_function() -> anyhow::Result<()> {
+    let abi_root = generate_abi_fn! {
+        pub fn add(&self, a: u32, b: u32) -> u32 {
+            a + b
+        }
+    };
+
+    assert_eq!(abi_root.abi.functions.len(), 1);
+    let function = &abi_root.abi.functions[0];
+    assert!(!function.is_init);
```

Comment:
why not just add this check in `test_view_function` to avoid duplication?

Reply:
Felt like splitting the tests makes each of the things I am trying to test stand out. But given that they are all very small and each test increases the run time quite a bit I agree that it makes sense to collapse some of them together